onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -189,6 +220,16 @@ class DmlOperatorQAttention : public DmlOperator
             }
         }
 
+        MLOperatorTensorDataType pastTensorDataType = MLOperatorTensorDataType::Undefined;
+        MLOperatorTensorDataType presentTensorDataType = MLOperatorTensorDataType::Undefined;
+        if (hasPast)
+        {
+            pastTensorDataType = kernelCreationContext.GetInputEdgeDescription(pastIndex).tensorDataType;
+            presentTensorDataType = kernelCreationContext.GetOutputEdgeDescription(presentIndex).tensorDataType;
+            auto pastTensorShape = m_inputTensorDescs[pastIndex].GetSizes();
+            m_inputTensorDescs[pastIndex] = TensorDesc::ConstructDefaultTensorDesc(kernelCreationContext.GetInputEdgeDescription(pastIndex).tensorDataType, pastTensorShape);
+        }
```

Comment:
I think `DmlOperator::Initialize` at line 114 will create the exact same tensorDesc for past tensor as here also we are creating default tensor desc with given data type and tensor shape. 
Can we please verify it?
If we end up need this creation here, then we can use above initialized `pastTensorDataType` variable in the method `TensorDesc::ConstructDefaultTensorDesc`.